compiler/next/include/chpl/queries/UniqueString.h

```diff
@@ -116,20 +117,16 @@ class UniqueString final {
     return this->s.i.v == other.s.i.v;
   }
   inline bool operator!=(const UniqueString other) const {
-    return this->s.i.v != other.s.i.v;
+    return !(*this == other);
   }
   int compare(const UniqueString other) const {
-    if (this->s.i.v == other.s.i.v)
-      return 0;
-    else
-      return strcmp(this->c_str(), other.c_str());
+    return *this == other ? 0 : strcmp(c_str(), other.c_str());
```

Comment:
I can go back to the original code, but I like the ternary operator for simple if-else value alternatives, similar to Python `x = 1 if y else 2`

It is more "functional", focusing on how a value is calculated, which I would think is more Chapel-esq.

I find it easier to read, since it's on one line and has a simple `cond ? this : that` syntax. But that's just my preference.

It does not affect compiler optimization.

Reply:
It can be made even simpler:
```
return *this == other ? 0 : compare(other.c_str());
```
If you don't need to make the `*this == other` case special, and can assume that `*this == other` implies that `strcmp(this->c_str(), other.c_str()) == 0`, then you can simply use:
```
return compare(other.c_str());
```
I think this is the simplest.

Reply:
I'm relatively happy with `return *this == other ? 0 : compare(other.c_str());` so go ahead with that please.
The `==` call is an optimization since it runs in constant time while strcmp is O(n).